### ISSUE TITLE
[Feature] 训练接口支持 temperature 参数

### ISSUE THREAD

**Issue by Etoilelune** [nex-agi/weaver#63]
## 背景

关联需求：china-qijizhifeng/nex-taas-feedback#144

训练侧需要支持 temperature scaling，用户需要能通过 SDK 指定训练时使用的 temperature，确保与推理侧一致。

## 使用场景

用户调用 `TrainingClient.forward_backward()`、`TrainingClient.compute_logprobs()` 等接口时，可以传入 temperature 参数，使训练侧的 logprob 计算使用 `softmax(logits / temperature)`。

## 预期行为

- `forward_backward()` 新增可选参数 `temperature: float = 1.0`
- `compute_logprobs()` 新增可选参数 `temperature: float = 1.0`
- 参数通过 REST API 传递给 server（china-qijizhifeng/weaver-server#263）
- 文档和类型注释同步更新
- 默认 temperature=1.0，向后兼容

## 关键验证方法

- SDK 传入 temperature 后，对应 API 请求中包含 temperature 参数
- 不传时默认 1.0，行为不变
- 端到端：SDK(temperature=0.7) → server → trainer → logprobs 使用 temperature=0.7 计算

**Comment by tang-t21** [comment]
Implemented by PR #66: https://github.com/nex-agi/weaver/pull/66
